editor/src/core/shared/project-file-types.ts

```diff
@@ -776,8 +776,20 @@ export type AllVariablesInScope = {
   variables: Variable[]
 }
 
+export type VariableUtopiaType =
+  | 'string'
+  | 'number'
+  | 'bigint'
+  | 'boolean'
+  | 'symbol'
+  | 'undefined'
+  | 'object'
+  | 'function'
+  | 'array'
+  | 'image'
+
 export interface Variable {
   name: string
-  type: 'string' | 'number' | 'bigint' | 'boolean' | 'symbol' | 'undefined' | 'object' | 'function'
+  type: VariableUtopiaType
```

Comment:
I think these types should be moved to `scoped-variables` , as they are only used in there and they feel more like internal representations than API. (not super strictly, but project-file-types.ts is about storing internal-api-level types)

I also don't like the name `VariableUtopiaType`, (again this is not a hard rule more like an observation) we try to not use Utopia to describe "custom" or "special-cased". If you move these into scoped-variables.ts then it becomes easier to name it, because there it can just be called DetectedType or InsertableType, you don't even have to export it, so the name is way less important (doesn't pollute global namespace)